src/parse.ts

```diff
@@ -125,7 +125,7 @@ export function stringifyParsedURL(parsed: Partial<ParsedURL>): string {
   const hash = parsed.hash || "";
   const auth = parsed.auth ? parsed.auth + "@" : "";
   const host = parsed.host || "";
-  const proto = parsed.protocol ? parsed.protocol + "//" : "";
+  const proto = parsed.protocol || host ? (parsed.protocol || "") + "//" : "";
```

Comment:
it is impossible to write this, which I think is clearer, due to conflicting linters:

```ts
const proto = parsed.protocol ? parsed.protocol + "//" : (host ? "//" : "");
```